orient_express/utils/retry.py

```
@@ -0,0 +1,45 @@
+
+    def decorator(old_func):
+        retry_conf = RetryConfig(retries, initial_timeout, max_timeout)
+
+        def new_func(*args, **kwargs):
+            timeout = retry_conf.initial_timeout
+            for i in range(retry_conf.retries):
+                try:
+                    return old_func(*args, **kwargs)
+                except Exception as e:
+                    logging.exception(
+                        f"Function {old_func.__name__} failed with error {e}, retry count {i}"
```

Ruff (0.14.10)

Redundant exception object included in `logging.exception` call

(TRY401)